packages/lib/services/profile.ts

```diff
@@ -0,0 +1,81 @@
+import { prisma } from "@formbricks/database";
+import { DatabaseError, ResourceNotFoundError } from "@formbricks/errors";
+import { Prisma } from "@prisma/client";
+import { TProfile, TProfileUpdateOutput } from "@formbricks/types/v1/profile";
+import { cache } from "react";
+
+const responseSelection = {
+  id: true,
+  name: true,
+  email: true,
+};
+
+// function to retrive basic information about a user's profile
+export const getProfile = cache(async (userId: string): Promise<TProfile> => {
+  try {
+    const profile = await prisma.user.findUnique({
+      where: {
+        id: userId,
+      },
+      select: responseSelection,
+    });
+
+    if (!profile) {
+      throw new ResourceNotFoundError("Profile", userId);
+    }
+
+    return profile;
+  } catch (error) {
+    if (error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new DatabaseError("Database operation failed");
+    }
+
+    throw error;
+  }
+});
+
+// function to update a user's profile
+export const updateProfile = async (
+  userId: string,
+  data: Prisma.UserUpdateInput
+): Promise<TProfileUpdateOutput> => {
+  try {
+    const currentProfile = await getProfile(userId);
```

Comment:
you don't need to get the current profile here. you can update the prisma update call so an ressourceNotFound error can be thrown in case the user was not found:

```ts
try {
  const updatedProfile = await prisma.user.update({
    where: {
      id: userId,
    },
    data: data,
  });

  // ... rest of your code

} catch (error) {
  if (error instanceof Prisma.PrismaClientKnownRequestError && error.code === 'P2016') {
    throw new ResourceNotFoundError("Profile", userId);
  } else {
    throw error;  // Re-throw any other errors
  }
}
```